### PR TITLE
use mixlib-cli's deprecation mechanism

### PR DESCRIPTION
**Pull request by marcparadise**
## Description

Deprecation logic is now available in `mixlib-cli`
via `deprecated_option`.  Let's use that instead
of keeping our custom deprecation behavior.

Signed-off-by: Marc A. Paradise <marc.paradise@gmail.com>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [ ] (na) I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -220,12 +220,12 @@ GEM
...
     mixlib-authentication (2.1.1)
-    mixlib-cli (2.0.6)
+    mixlib-cli (2.1.0)
     mixlib-config (3.0.1)
       tomlrb
     mixlib-log (3.0.1)
-    mixlib-shellout (2.4.4)
-    mixlib-shellout (2.4.4-universal-mingw32)
+    mixlib-shellout (3.0.4)
+    mixlib-shellout (3.0.4-universal-mingw32)
```
Comment (lamont-granquist) [inline Gemfile.lock]: and mixlib-shellout 3.0.4 snuck in 👍

**Comment by marcparadise** [comment]
Looking into the test failures. The two in application_spec are related.

**Comment by marcparadise** [comment]
The test failure is resolved with https://github.com/chef/mixlib-cli/pull/66

**Comment by marcparadise** [comment]
kitchen failures were caused by a mis-tag of ohai 15.0.39.  I created tag v15.0.39 which should resolve that.

**Pull request merged**